assets/javascript/progressive-reveal.js

```diff
@@ -4,68 +4,66 @@ var each = require('lodash.foreach');
 var groupBy = require('lodash.groupby');
 
 var helpers = require('./helpers'),
-    inputs, groups,
-    toggleAttr = 'data-toggle',
-    hiddenClass = 'js-hidden';
+  inputs, groups,
+  toggleAttr = 'data-toggle',
+  hiddenClass = 'js-hidden';
 
 function inputClicked(e, target) {
-    target = target || helpers.target(e);
-    var shown;
-    each(groups[target.name], function (input) {
-        var id = input.getAttribute('aria-controls');
-        var toggle = document.getElementById(id);
-        if (toggle) {
-            if (input.checked) {
-                input.setAttribute('aria-expanded', 'true');
-                toggle.setAttribute('aria-hidden', 'false');
-                helpers.removeClass(toggle, hiddenClass);
-                shown = toggle.id;
-                var elementToFocus = toggle.getElementsByTagName('input')[0];
-                if ((input === target) && elementToFocus) {
-                    elementToFocus.focus();
-                }
-            } else {
-                input.setAttribute('aria-expanded', 'false');
-                if (shown !== toggle.id) {
-                    toggle.setAttribute('aria-hidden', 'true');
-                    helpers.addClass(toggle, hiddenClass);
-                }
-            }
+  target = target || helpers.target(e);
+  var shown;
+  each(groups[target.name], function (input) {
+    var id = input.getAttribute('aria-controls');
+    var toggle = document.getElementById(id);
+    if (toggle) {
+      if (input.checked) {
+        toggle.setAttribute('aria-hidden', 'false');
+        helpers.removeClass(toggle, hiddenClass);
+        shown = toggle.id;
+        var elementToFocus = toggle.querySelectorAll('input,textarea')[0];
```

Comment:
Instead of getElementsByTagName('input')[0], we now also find textarea

Reply:
Could there be any knock on impact here that getElementsByTagName returns a live HTMLCollection, whereas  Element.quertSelectorAll() returns a static NodeList? Not sure regarding the context that this is used given there are progressive revealing of items.

Reply:
I think in this context it's okay because it just selects the item. Both collections support the array element accessor style. From what I gather the HTMLCollection is live and will update itself as the DOM is manipulated. But as we're just immediately focus on item after finding it, it won't make a difference.